velox/exec/DistinctAggregations.cpp

```diff
@@ -185,6 +199,47 @@ class TypedDistinctAggregations : public DistinctAggregations {
     return input->template asUnchecked<RowVector>()->children();
   }
 
+  void extractForSpill(folly::Range<char**> groups, VectorPtr& result) const {
+    auto* arrayVector = result->as<ArrayVector>();
+    arrayVector->resize(groups.size());
+
+    auto* rawOffsets =
+        arrayVector->mutableOffsets(groups.size())->asMutable<vector_size_t>();
+    auto* rawSizes =
+        arrayVector->mutableSizes(groups.size())->asMutable<vector_size_t>();
+
+    size_t totalBytes = 0;
+    vector_size_t offset = 0;
+    for (auto i = 0; i < groups.size(); ++i) {
+      auto* accumulator =
+          reinterpret_cast<AccumulatorType*>(groups[i] + offset_);
+      rawSizes[i] = accumulator->maxSpillSize();
+      rawOffsets[i] = offset;
+      offset += accumulator->maxSpillSize();
+      totalBytes += accumulator->maxSpillSize();
+    }
+
+    auto& elementsVector = arrayVector->elements();
+    elementsVector->resize(totalBytes);
+
+    auto* flatVector = arrayVector->elements()->asFlatVector<StringView>();
+    flatVector->resize(1);
+    auto* rawBuffer = flatVector->getRawStringBufferWithSpace(totalBytes, true);
```

Comment:
I'm too confused to continue reading. Will wait for you to answer the questions. Thanks.

Reply:
Answered the questions - glad to change as needed to make the intent clearer. Sorry, first time using these FlatVector constructs, so I may be abusing them.